app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/continuousdelivery/ContinuousDeliveryHandler.java

```diff
@@ -0,0 +1,184 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.endpoint.v1.handler.continuousdelivery;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.StreamingOutput;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import io.syndesis.common.model.connection.ConnectionOverview;
+import io.syndesis.common.model.integration.ContinuousDeliveryEnvironment;
+import io.syndesis.common.model.integration.ContinuousDeliveryImportResults;
+import io.syndesis.common.model.integration.IntegrationDeployment;
+import io.syndesis.common.model.monitoring.IntegrationDeploymentStateDetails;
+
+@Api("continuous-delivery")
+@Path("/public/continuousdelivery")
+public interface ContinuousDeliveryHandler {
+
+    /**
+     * List all available environments.
+     */
+    @GET
+    @Path("environments")
+    @Produces(MediaType.APPLICATION_JSON)
+    List<String> getReleaseEnvironments();
+
+    /**
+     * List all tags associated with this integration.
+     */
+    @GET
+    @Path("releasetags/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    Map<String, ContinuousDeliveryEnvironment> getReleaseTags(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId);
+
+    /**
+     * Delete an environment tag associated with this integration.
+     */
+    @DELETE
+    @Path("releasetags/{id}/{env}")
+    void deleteReleaseTag(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId, @NotNull @PathParam("env") @ApiParam(required = true) String environment);
+
+    /**
+     * Tag an integration for release to target environments.
+     */
+    @POST
+    @Path("releasetags/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    Map<String, ContinuousDeliveryEnvironment> tagForRelease(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId,
+                                                @NotNull @ApiParam(required = true) List<String> environments);
+
+    /**
+     * Export integrations to a target environment.
+     */
+    @GET
+    @Path("export/{env}/export.zip")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    StreamingOutput exportResources(@NotNull @PathParam("environment") @ApiParam(required = true) String environment,
```

Comment:
Is this needed?

Reply:
Used as a convenience filter for simple CD scripts in the first version.